lib/elixir/lib/file.ex

```diff
@@ -791,35 +795,38 @@ defmodule File do
   Returns `:ok` otherwise.
   """
   @spec cp!(Path.t(), Path.t(), (Path.t(), Path.t() -> boolean)) :: :ok
-  def cp!(source, destination, callback \\ fn _, _ -> true end) do
-    case cp(source, destination, callback) do
+  def cp!(source_file, destination_file, callback \\ fn _, _ -> true end) do
+    case cp(source_file, destination_file, callback) do
       :ok ->
         :ok
 
       {:error, reason} ->
         raise File.CopyError,
           reason: reason,
           action: "copy",
-          source: IO.chardata_to_string(source),
-          destination: IO.chardata_to_string(destination)
+          source: IO.chardata_to_string(source_file),
+          destination: IO.chardata_to_string(destination_file)
     end
   end
 
   @doc ~S"""
-  Copies the contents in source to destination.
+  Copies the contents in `source` to `destination` recursively, maintaining the
+  source directory structure and modes.
```

Comment:
Here I talk about "modes" which was omitted before, but since this function relies on `do_cp_file/4` which is the same function `cp/3` relies on, it is assumed to keep modes, with only one mention bellow (which is a bug IMO)